spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanTracker.scala

```diff
@@ -73,29 +74,45 @@ class SparkExecutionPlanTracker(
     while (!stopped) {
       try {
         Option(qeQueue.poll(3000, TimeUnit.MILLISECONDS)).foreach { qd =>
-          val entities = qd.qe.sparkPlan.collect { case p: LeafExecNode => p }
-            .flatMap {
+          val entities = qd.qe.sparkPlan.collect {
+            case p: DataWritingCommandExec => p
+            case p: LeafExecNode => p
+          }.flatMap {
             case r: ExecutedCommandExec =>
               r.cmd match {
-                case c: InsertIntoHiveTable =>
-                  logDebug(s"INSERT query ${qd.qe}")
-                  CommandsHarvester.InsertIntoHiveTableHarvester.harvest(c, qd)
-
-                // Case 6. CREATE TABLE AS SELECT
-                case c: CreateHiveTableAsSelectCommand =>
-                  logDebug(s"CREATE TABLE AS SELECT query: ${qd.qe}")
-                  CommandsHarvester.CreateHiveTableAsSelectHarvester.harvest(c, qd)
-
                 case c: LoadDataCommand =>
                   // Case 1. LOAD DATA LOCAL INPATH (from local)
                   // Case 2. LOAD DATA INPATH (from HDFS)
                   logDebug(s"LOAD DATA [LOCAL] INPATH (${c.path}) ${c.table}")
                   CommandsHarvester.LoadDataHarvester.harvest(c, qd)
 
+                // Case 6. CREATE TABLE AS SELECT
+                case c: CreateHiveTableAsSelectCommand =>
+                  logDebug(s"CREATE TABLE AS SELECT query: ${qd.qe}")
+                  CommandsHarvester.CreateHiveTableAsSelectHarvester.harvest(c, qd)
+
                 case c: CreateDataSourceTableAsSelectCommand =>
                   logDebug(s"CREATE TABLE USING xx AS SELECT query: ${qd.qe}")
                   CommandsHarvester.CreateDataSourceTableAsSelectHarvester.harvest(c, qd)
 
+                case c: CreateViewCommand =>
```

Comment:
Do we have a test case for this?

Reply:
Ohh, not yet. I'll add unit tests for CreateViewCommand